pkg/catalog/routes.go

```diff
@@ -397,3 +418,75 @@ func listTrafficTargetPermutations(name string, srcServiceList []service.MeshSer
 
 	return trafficPolicies
 }
+
+func (mc *MeshCatalog) getWeightedBackendsForService(svc service.MeshService) []service.WeightedService {
+	weightedSvcs := []service.WeightedService{}
+
+	// Retrieve the weighted clusters from traffic split
+	// Assumes no nested splits for now
+	// Assumes service would only be root once (not sure if this is validated)
+	servicesList := mc.meshSpec.ListTrafficSplitServices()
+	for _, activeService := range servicesList {
+		if activeService.RootService == svc.Name {
+			weightedSvcs = append(weightedSvcs, activeService)
+		}
+	}
+
+	if len(weightedSvcs) == 0 {
+		svc := service.WeightedService{
+			Service: svc,
+			Weight:  constants.ClusterWeightAcceptAll,
+		}
+		weightedSvcs = append(weightedSvcs, svc)
+	}
+	return weightedSvcs
+}
+
+func (mc *MeshCatalog) getRouteWeightedServices(src service.MeshService, dest service.MeshService, trafficTargets map[string]trafficpolicy.TrafficTarget) []trafficpolicy.RouteWeightedServices {
+	routeWeightedSvcs := []trafficpolicy.RouteWeightedServices{}
+
+	weightedBackends := mc.getWeightedBackendsForService(dest)
+
+	routeServicesMap := make(map[*trafficpolicy.HTTPRoute][]service.WeightedService)
+
+	for _, backend := range weightedBackends {
+		hash := hashSrcDstService(src, backend.Service)
+		routes := trafficTargets[hash].HTTPRoutes
```

Comment:
shouldn't `trafficTargets[hash]` be safeguarded and check it exists before accessing an element into it?
even if it were to be guaranteed by some other precondition, best practices to err check